app/main/views/suppliers.py

```diff
@@ -439,9 +439,18 @@ def do_search(search_query, offset, result_count, new_domains, framework_slug):
             q = q.having(sr_agg.contains(array(roles_list)))
 
     if seller_types_list is not None and 'recruiter' in seller_types_list:
-        q = q.filter(Supplier.is_recruiter == 'true')
+        q = q.filter(Supplier.data['recruiter'].astext == 'yes')
         seller_types_list.remove('recruiter')
-        if len(seller_types_list) == 0:
+
+    if seller_types_list is not None and 'consultant_and_recruiter' in seller_types_list:
+        q = q.filter(Supplier.data['recruiter'].astext == 'both')
+        seller_types_list.remove('consultant_and_recruiter')
+
+    if seller_types_list is not None and 'consultant' in seller_types_list:
+        q = q.filter(Supplier.is_recruiter == 'false')
```

Comment:
I used `is_recruiter` filter since that would take into account legacy sellers that do not have a `recruiter `key or an empty recruiter string in their json data.

I thought it is best to use `is_recruiter ` as a filter rather than combining both filters for instance `q = q.filter(Supplier.data['recruiter'].astext == 'no' or Supplier.is_recruiter == 'false )` since that would be unnecessary. 

We don't need to add this logic for pure recruiters since **PROD** data does not have any legacy sellers that are recruiters or hybrids. 
The seller filter would be an issue in the testing environment.